src/main.rs

```diff
@@ -161,6 +167,23 @@ fn check_host(target: &CanaryTarget) -> CanaryCheck {
     }
 }
 
+
+/*
+    Checks if alert would not be spam.
+    Alert would not be spam iff state changes from "okay" to "fire" or "fire" to "okay".
+    Updates HashMap last_statuses with last state seen for the target.
+*/
+fn check_not_spam(last_statuses: &mut HashMap<CanaryTarget, String>, result: &CanaryCheck) -> bool {
+    let not_spam = match last_statuses.get(&result.target) {
+        Some(status) => {
+            (status == "okay" && result.info == "fire") || (status == "fire" && result.info == "okay")
```

Comment:
I am not sure of the expected behavior now. The initial issue suggested that only changes from `Ok` to `Fire` and `Fire` to `Ok` are valid. Should the behavior instead be that *any* status change is not spam?